STYLE_GUIDE.md

```diff
@@ -0,0 +1,185 @@
+
+Avoid:
+```scss
+color: #001521;
+color: rgba(#001521, 0.5);
+color: rgba(0, 0, 0, 0.5);
+```
+
+Prefer:
+```scss
+color: $color-bg-dark-1;
+color: rgba($color-bg-dark-1, 0.5);
+color: rgba(black, 0.5);
+```
+
+### Theme colours
+
+Kedro-Viz has both light and dark themes, and every component must be styled appropriately for each theme. The themes are set using the same theme classes that Kedro-UI uses - `kui-theme--light` and `kui-theme--dark` - applied on the top-level container element. In order to simplify code and avoid increasing CSS specificity, we use CSS Custom Properties to handle theme colours, and we recommend that you use these where possible.
+
+Avoid:
+```scss
+.pipeline-title {
+  .kui-theme--light & {
+    background: $color-bg-light-1;
+  }
+
+  .kui-theme--dark & {
+    background: $color-bg-dark-1;
+  }
+}
+```
+
+Prefer:
+```scss
+.kui-theme--light {
+  --color-bg-1: $color-bg-light-1;
+}
+
+.kui-theme--dark {
+  --color-bg-1: $color-bg-dark-1;
+}
+
+.pipeline-title {
+  background: var(--color-bg-1);
+}
+```
+
+<!-- @TODO -->
+
+### Units (em, %, px, etc)
+
+<!-- @TODO -->
+
+## JavaScript
+
+### Imports
+
+Although it will work in regular development (because this project uses React-Scripts), avoid importing non-standard file-types like `.scss` and `.svg` directly into JavaScript files. This is because these won't work without specific webpack loaders, so when Kedro-Viz is imported into other projects, it will break. The `lib-test` testing suite exists partly for this reason, to check that Kedro-Viz still works when imported as a componnent library into a fairly standard JS app.
+
+###  Testing
+
+In accordance with McKinsey standards, we aim to maintain at least a test coverage of at least 70% averaged across the codebase. As it's not practical to write tests for every JavaScript file, we usually aim for closer to 90% on most files, so that the project average continues to exceed the expected minimum standard.
+
+We use Jest, Enzyme, and React-Testing-Library for JavaScript testing. Most of the older tests are written with Enzyme, but we are beginning to write more tests using Testing-Library as it is more flexible for testing certain browser APIs. Either is acceptable. To help with mocking the Redux store, Enzyme helper utilities can be found in `/src/utils/state.mock.js`.
+
+<!-- @TODO Explain the usage of the different datasets in testing, and add more comments explaining how the load-data mock works. -->
+
+## Git and Github
+
+### Commits
+
+Write commit subjects/titles in the imperative mood. Try to be as specific and detailed as you can, while keeping the subject under 50 characters so that isn't truncated when displayed. If it's a complicated or unclear change and more detail would help future readers understand the commit, please consider adding a detailed description of what you changed and why.
```

Comment:
```suggestion
Write commit subjects/titles in the imperative mood. Try to be as specific and detailed as you can, yet keep the subject under 50 characters so that isn't truncated when displayed. If it's a complicated or unclear change and more detail would help future readers understand the commit, add a detailed description of what you changed and why.
```